raiden/transfer/views.py

```diff
@@ -638,6 +638,41 @@ def filter_channels_by_partneraddress(
     return result
 
 
+def filter_channels_to_settle_by_partner_address(
+    chain_state: ChainState,
+    payment_network_id: PaymentNetworkID,
+    token_address: TokenAddress,
+    partner_addresses: List[Address],
+) -> List[NettingChannelState]:
+    token_network = get_token_network_by_token_address(
+        chain_state, payment_network_id, token_address
+    )
+
+    result: List[NettingChannelState] = []
+    if not token_network:
+        return result
+
+    channelsResult = []
+    # for partner in partner_addresses:
+
+    for node_address in token_network.partneraddresses_to_channelidentifiers.keys():
+        if node_address in token_network.channelidentifiers_to_channels:
+            channels = token_network.channelidentifiers_to_channels[node_address]
+            if channels is not None:
+                for _channelId, channel in channels.items():
+                    for partner_address in partner_addresses:
+                        if channel.partner_state.address == partner_address:
+                            if channel.close_transaction and channel.close_transaction.result == 'success':
```

Comment:
I think this could be  refactorized in order to prevent such amount of identation levels.

Reply:
We can change that into this:

```
def filter_channels_to_settle_by_partner_address(
    chain_state: ChainState,
    payment_network_id: PaymentNetworkID,
    token_address: TokenAddress,
    partner_addresses: List[Address],
) -> List[NettingChannelState]:
    token_network = get_token_network_by_token_address(
        chain_state, payment_network_id, token_address
    )

    result: List[NettingChannelState] = []
    if not token_network:
        return result

    node_addresses = filter(lambda node_address: node_address in token_network.channelidentifiers_to_channels,
                            token_network.partneraddresses_to_channelidentifiers.keys())
    channels = filter(lambda channels: channels is not None,
                      map(lambda node_address: token_network.channelidentifiers_to_channels[node_address],
                          node_addresses))

    channels_result = map(lambda partner_address: filter(lambda _channelId, channel: channel.partner_state.address == partner_address and
                                                         channel.close_transaction and
                                                         channel.close_transaction.result == 'success',
                             channels.items()), partner_addresses)

    states = filter_channels_by_status(channels_result, [CHANNEL_STATE_SETTLING])
    # If multiple channel states are found, return the last one.
    if states:
        result.append(states[-1])

    return result
```
Tell me what do you think about it, maybe we can analize to change that into lambda expresions like this

Reply:
Finally we removed this so

Reply:
I like it.

Anyway if the code is in other places, I think it's not urgent to modifiy it, but it would be nice.